src/Sulu/Bundle/ContentBundle/Search/EventSubscriber/BlameTimestampSubscriber.php

```diff
@@ -0,0 +1,201 @@
+use Massive\Bundle\SearchBundle\Search\Event\PreIndexEvent;
+use Sulu\Bundle\SearchBundle\Search\Document;
+use Sulu\Component\Security\Authentication\UserInterface;
+use Sulu\Bundle\SecurityBundle\Entity\User;
+use Massive\Bundle\SearchBundle\Search\Event\HitEvent;
+
+/**
+ * Add blame (creator, changor) and timestamp (created, changed) to
+ * the document before it is indexed.
+ *
+ * Works for both objects implementing interfaces (UserBlameInterface and
+ * TimestampableInterface) and Structures.
+ */
+class BlameTimestampSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var Factory
+     */
+    private $factory;
+
+    /**
+     * @var EntityManager
+     */
+    private $entityManager;
+
+    /**
+     * @param Factory $factory
+     */
+    public function __construct(Factory $factory, EntityManager $entityManager)
+    {
+        $this->factory = $factory;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            SearchEvents::PRE_INDEX => 'handleBlameTimestamp',
+            SearchEvents::HIT => 'handleBlameTimestampHitMapping',
+        );
+    }
+
+    /**
+     * Map blame and timestamp information to the search document
+     *
+     * @param PreIndexEvent $event
+     */
+    public function handleBlameTimestamp(PreIndexEvent $event)
+    {
+        $subject = $event->getSubject();
+        $document = $event->getDocument();
+
+        if ($subject instanceof UserBlameInterface) {
+            $this->mapCreatorAndChanger($document, $subject->getCreator(), $subject->getChanger());
+        }
+
+        if ($subject instanceof TimestampableInterface) {
+            $this->mapTimestamp($document, $subject->getCreated(), $subject->getChanged());
+        }
+
+        if ($subject instanceof StructureInterface) {
+            $this->handleStructure($subject, $document);
+        }
+    }
+
+    /**
+     * Map the changer and created from the field data to
+     * the search document (we don't include the field data in the search result API)
+     *
+     * @param HitEvent $event
+     */
+    public function handleBlameTimestampHitMapping(HitEvent $event)
+    {
+        $document = $event->getHit()->getDocument();
+        $this->doHandleBlameTimestampHitMapping($document);
+    }
+
+    /**
+     * @param Document $document
+     */
+    private function doHandleBlameTimestampHitMapping(Document $document)
+    {
+        $document->setCreatorName($this->getFieldValue($document, 'creator'));
+        $document->setChangerName($this->getFieldValue($document, 'changer'));
+        $document->setCreatorId($this->getFieldValue($document, 'creator_id'));
+        $document->setChangerId($this->getFieldValue($document, 'changer_id'));
+        $document->setCreated($this->getFieldValue($document, 'created'));
+        $document->setChanged($this->getFieldValue($document, 'changed'));
```

Comment:
Hmm, but then again -- there is a single Document object for the Sulu integeration. All Sulu entities have the above fields. So it makes sense that this information is part of the Document API. Above we map this information from the documents fields to be properties of the document.